### ISSUE TITLE
Yield inconsistency

### ISSUE THREAD

**Issue by kylekatarnls** [pugjs/pug#2878]
**Pug Version:** 2.0.0-rc.4

**Node Version:** 7.7.1

## Input Pug

input.pug
```pug
include a.pug
  | b
```

## Expected HTML

I expect the same output for each a.pug code in the following table

|a.pug|Actual HTML|
|----|----|
|`p a`|`<p>ab</p>`|
|`p.\n  a`|`<p>a</p>b`|
|`p\n  \|a`|`<p>a</p>b`|

`\n` are new lines.

## Additional Comments

The results above IMHO should not be different, it seems to me like an inconsistency. Else what is the reason for?

This is the last test that fails on the PHP port https://github.com/pug-php/pug/pull/162 to get full compliance and release our new version but before to implement this the same way in our port, I would be sure it's not a bug.

Thanks,

**Comment by batiste** [comment]
Yes I just came across this very weird behavior. Here is another example:

/components/blop
```pug
1

2
  | 

3
```


```pug
include /components/blop
  | test
```

Result

```html
<1></1>
<2> test</2>
<3></3>
```

What is the weird algorithm for inserting the default block? Also shouldn't we be able to use block with mixins and include. I tried to use them but it seems the block are somehow completely ignored?

**Comment by batiste** [comment]
Another example:

components/blip
```pug
div
  1

div
  2
  block test2
  block test1

div
  3
```

```
include /components/blip
  block test1
    t1
  block test2
    t2

```

Gives you the completely unpredictable:

```html
<div>
  <1></1>
</div>
<div>
  <2></2>
</div>
<div>
  <3></3>
  <t1></t1>
  <t2></t2>
</div>
```

**Comment by kylekatarnls** [comment]
@batiste You found out a very strange case for your first example.

If I well understood, you have in pugjs 2:

- named block (`block test1`) that works only with extends/layout.
- anonymous block (`block`) for mixins.
- `yield` for includes. (This can only be anonymous I think)

So you can get a similar mechanism that what you describe in your second example by combining extend, named block, include and yield. I will try to make a working exemple.

**Comment by batiste** [comment]
@kylekatarnls I managed to use yield with include but how would you use anonymous blocks with mixin? Can you have several?

**Comment by batiste** [comment]
Oki so here the syntax for the unamed block for a mixin is well documented here:

https://pugjs.org/language/mixins.html#mixin-blocks

**Comment by kylekatarnls** [comment]
@ForbesLindesay Hi, any news about this?
